modules/UI/videolayout/VideoLayout.js

```diff
@@ -405,20 +406,17 @@ var VideoLayout = {
                 this.updateLargeVideo(this.electLastVisibleVideo());
             }
 
-            eventEmitter.emit(UIEvents.PINNED_ENDPOINT, smallVideo, false);
-
             return;
         }
 
-        // Lock new video
-        pinnedId = id;
-
         // Update focused/pinned interface.
         if (id) {
-            if (smallVideo && !interfaceConfig.filmStripOnly)
+            if (smallVideo && !interfaceConfig.filmStripOnly) {
                 smallVideo.focus(true);
-
-            eventEmitter.emit(UIEvents.PINNED_ENDPOINT, smallVideo, true);
+                APP.store.dispatch(pinParticipant(id));
+            }
+        } else {
+            APP.store.dispatch(pinParticipant(null));
```

Comment:
I don't think this is a valid case but update the internal state of the pinnedId here to be safe.